### PR TITLE
Add Acceptance tests for Analytics [MAILPOET-5405]

### PR DESCRIPTION
**Pull request by websupporter**
## Description
Adds the tests and smaller bugfixes.

Bugfixes:
- Show order related stuff only when WC is active
- When the open rate/click rate/revenue etc in the previous period was higher, we need to show `-x%` instead of `x%` in the Overview Panel

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/752

## Linked tickets

[MAILPOET-5405]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5405]: https://mailpoet.atlassian.net/browse/MAILPOET-5405?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/tests/DataFactories/AutomationRun.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    $automationRun = new Entity(
+      $this->automation->getId(),
+      $this->automation->getVersionId(),
+      $this->triggerKey,
+      $this->subjects
+    );
+
+    $id = $this->storage->createAutomationRun($automationRun);
+    $automationRun = $this->storage->getAutomationRun($id);
+    $this->storage->deleteAutomationRun($automationRun);
```
Comment (JanJakes) [inline mailpoet/tests/DataFactories/AutomationRun.php]: Why does this create and immediately delete an automation run? 🤔

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx [hunk 2]
```diff
@@ -39,15 +39,16 @@ function getEmailDelta(type: 'opened' | 'clicked'): number | undefined {
...
   }
 
   if (previous === 0) {
     return 0;
   }
 
   const newValue = current > previous ? current - previous : previous - current;
-  return (newValue / previous) * 100;
+  const delta = (newValue / previous) * 100;
+  return current > previous ? delta : delta * -1;
```
Comment (JanJakes) [inline mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx]: I guess the original `current > previous ? current - previous : previous - current` is not needed and then it's simply:

```tsx
const newValue = current - previous;
return (newValue / previous) * 100;
```

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx [hunk 3]
```diff
@@ -80,7 +80,8 @@ function getWooCommerceDelta(type: 'revenue' | 'orders'): number | undefined {
     return 0;
   }
 
-  return (newValue / previous) * 100;
+  const delta = (newValue / previous) * 100;
+  return current > previous ? delta : delta * -1;
```
Comment (JanJakes) [inline mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx]: I guess the original `current > previous ? current - previous : previous - current` is not needed and then it's simply:

```tsx
const newValue = current - previous;
return (newValue / previous) * 100;
```

File: mailpoet/tests/DataFactories/WooCommerceOrder.php [hunk 4]
```diff
@@ -95,16 +95,37 @@ public function create() {
...
+   */
+  private function updateOrderCreateDateWhenWooIsNotAvailable(int $orderId, string $date) {
+    $cmd = ['post', 'update', '--porcelain', $orderId ];
+    $cmd[] = '--post_date="' . $date . '"';
+    $cmd[] = '--post_date_gmt="' . $date . '"';
+    $cmd[] = '--post_modified="' . $date . '"';
+    $cmd[] = '--post_modified_gmt="' . $date . '"';
+    $this->tester->cliToString($cmd);
+  }
+
```
Comment (JanJakes) [inline mailpoet/tests/DataFactories/WooCommerceOrder.php]: Why would we need to update an order without Woo? 🤔

Comment (JanJakes) [reply mailpoet/tests/DataFactories/WooCommerceOrder.php]: Ahh, fixed in https://github.com/mailpoet/mailpoet-premium/pull/752/commits/a7878e83fba12a93a778b4d4e7d86b548e945124.

File: mailpoet/tests/DataFactories/AutomationRun.php (hunk 1 above)
Comment (JanJakes) [reply mailpoet/tests/DataFactories/AutomationRun.php]: Fixed in https://github.com/mailpoet/mailpoet/pull/5065/commits/f7f46c57c6bf0df1db31dfa125c1481b605e701a.

**Review by JanJakes — APPROVED** [review]

**Review by veljkho — APPROVED** [review]

**Pull request merged**